weblate/templates/dictionary.html

```diff
@@ -84,6 +84,11 @@
 {% else %}
 <p>{% trans "No words found!" %}</p>
 {% endif %}
+<form class="searchform" action="{% url 'show_dictionary' project=project.slug lang=language.code %}">
+{% csrf_token %}
+<input name="term" value="{{ search }}"/>
```

Comment:
You should render the form here instead of manually creating the markup. Probably just '{{ searchform }}' will do that.